mesonbuild/interpreter.py

```diff
@@ -2229,7 +2250,12 @@ def do_subproject(self, dirname, kwargs):
             incpath = ' => '.join(fullstack)
             raise InvalidCode('Recursive include of subprojects: %s.' % incpath)
         if dirname in self.subprojects:
-            return self.subprojects[dirname]
+            subproject = self.subprojects[dirname]
+
+            if required and not subproject.found():
+                raise InterpreterException('Subproject "%s" required but not found.' % (dirname))
```

Comment:
We need to print `subproject_dir/dirname` in error messages, same as below when we print that a subproject does not exist.

Reply:
Fixed.